crates/bevy_text/src/input.rs

```diff
@@ -0,0 +1,1312 @@
+        let value = value.0.clone();
+        if let Some(mut actions) = world.entity_mut(context.entity).get_mut::<TextEdits>() {
+            actions.queue(TextEdit::SetText(value));
+        }
+    }
+}
+
+/// The time taken for the cursor to blink
+#[derive(Resource)]
+pub struct TextCursorBlinkInterval(pub Duration);
+
+impl Default for TextCursorBlinkInterval {
+    fn default() -> Self {
+        Self(Duration::from_secs_f32(0.5))
+    }
+}
+
+/// Common text input properties set by the user.
+/// On changes, the text input systems will automatically update the buffer, layout and fonts as required.
+#[derive(Component, Debug, PartialEq)]
+pub struct TextInputAttributes {
+    /// The text input's font, also used for any [`Placeholder`] text or password mask.
+    /// A text input's glyphs must all be from the same font.
+    pub font: Handle<Font>,
+    /// The size of the font.
+    /// A text input's glyphs must all be the same size.
+    pub font_size: f32,
+    /// The height of each line.
+    /// A text input's lines must all be the same height.
+    pub line_height: LineHeight,
+    /// Determines how lines will be broken
+    pub line_break: LineBreak,
+    /// The horizontal alignment for all the text in the text input buffer.
+    pub justify: Justify,
+    /// Controls text antialiasing
+    pub font_smoothing: FontSmoothing,
+    /// Maximum number of glyphs the text input buffer can contain.
+    /// Any edits that extend the length above `max_chars` are ignored.
+    /// If set on a buffer longer than `max_chars` the buffer will be truncated.
+    pub max_chars: Option<usize>,
+    /// The maximum number of lines the buffer will display without scrolling.
+    /// * Clamped between zero and target height divided by line height.
+    /// * If None or equal or less than 0, will fill the target space.
+    /// * Only restricts the maximum number of visible lines, places no constraint on the text buffer's length.
+    /// * Supports fractional values, `visible_lines: Some(2.5)` will display two and a half lines of text.
+    pub visible_lines: Option<f32>,
+    /// Clear on submit (Triggered when [`apply_text_edits`] receives a [`TextEdit::Submit`] edit for an entity).
+    pub clear_on_submit: bool,
+}
+
+impl Default for TextInputAttributes {
+    fn default() -> Self {
+        Self {
+            font: Default::default(),
+            font_size: 20.,
+            line_height: LineHeight::RelativeToFont(1.2),
+            font_smoothing: Default::default(),
+            justify: Default::default(),
+            line_break: Default::default(),
+            max_chars: None,
+            visible_lines: None,
+            clear_on_submit: false,
+        }
+    }
+}
+
+/// If a text input entity has a `TextInputFilter` component, after each [`TextEdit`] is applied, the [`TextInputBuffer`]’s text is checked
+/// against the filter, and if it fails, the `TextEdit` is rolled back.
```

Comment:
Is this immediately rolled back, or is invalid state possible? Either way this needs to be clearly documented.